libkbfs/key_manager.go

```diff
@@ -454,14 +456,16 @@ func (km *KeyManagerStandard) Rekey(ctx context.Context, md *RootMetadata, promp
 	defer func() { km.deferLog.CDebugf(ctx, "Rekey %s done: %+v", md.TlfID(), err) }()
 
 	currKeyGen := md.LatestKeyGeneration()
-	if md.TlfID().IsPublic() != (currKeyGen == PublicKeyGen) {
+	if (md.TlfID().Type() == tlf.Public) != (currKeyGen == PublicKeyGen) {
```

Comment:
The "clever" factor here is high enough that it warrants a comment, something like "inconsistent indicators that this TLF is or isn't public".